File: backend/LendACarAPI/DTOs/EmployeeDto.cs
﻿using LendACarAPI.Data.Models;
using System.ComponentModel.DataAnnotations;

namespace LendACarAPI.DTOs
{
    public class EmployeeDto
    {
        public int Id { get; set; }

        [Required]
        public string? FirstName { get; set; }

        [Required]
        public string? LastName { get; set; }

        [Required]
        [Phone]
        public string? PhoneNumber { get; set; }

        [Required]
        public string BirthDate { get; set; }

        public int CityId { get; set; }
        public City? City { get; set; }

        [Required]
        [EmailAddress]
        public string? EmailAddress { get; set; }

        [Required]
        public string? Username { get; set; }

        public int WorkingHourId { get; set; }
        public WorkingHour? WorkingHour { get; set; }

        public string? JobTitle { get; set; }
    }
}

File: frontend/src/app/Models/EmployeeDto.ts
import {City} from './City';
import {WorkingHour} from './WorkingHour';

export interface EmployeeDto {
  id:number;
  firstName: string;
  lastName: string;
  phoneNumber: string;
  birthDate: string;
  cityId: number;
  city: City;
  emailAddress: string;
  username: string;
  workingHour:WorkingHour
  workingHourId:number
  jobTitle: string;
}
